test/client/snapshot-request.js

```diff
@@ -0,0 +1,386 @@
+      data: {
+        title: 'Don Quixote',
+        author: 'Miguel de Cervantes'
+      }
+    };
+
+    beforeEach(function (done) {
+      var doc = backend.connect().get('books', 'don-quixote');
+      doc.create({ title: 'Don Quixote' }, function (error) {
+        if (error) return done(error);
+        clock.tick(ONE_DAY);
+        doc.submitOp({ p: ['author'], oi: 'Miguel de Cervante' }, function (error) {
+          if (error) return done(error);
+          clock.tick(ONE_DAY);
+          doc.submitOp({ p: ['author'], od: 'Miguel de Cervante', oi: 'Miguel de Cervantes' }, done);
+        });
+      });
+    });
+
+    it('fetches v0', function (done) {
+      backend.connect().getSnapshot('books', 'don-quixote', 0, function (error, snapshot) {
+        if (error) return done(error);
+        expect(snapshot).to.eql(v0);
+        done();
+      });
+    });
+
+    it('fetches v1', function (done) {
+      backend.connect().getSnapshot('books', 'don-quixote', 1, function (error, snapshot) {
+        if (error) return done(error);
+        expect(snapshot).to.eql(v1);
+        done();
+      });
+    });
+
+    it('fetches v2', function (done) {
+      backend.connect().getSnapshot('books', 'don-quixote', 2, function (error, snapshot) {
+        if (error) return done(error);
+        expect(snapshot).to.eql(v2);
+        done();
+      });
+    });
+
+    it('fetches the version from Day 1', function (done) {
+      backend.connect().getSnapshot('books', 'don-quixote', DAY1, function (error, snapshot) {
+        if (error) return done(error);
+        expect(snapshot).to.eql(v0);
+        done();
+      });
+    });
+
+    it('fetches the version from Day 2', function (done) {
+      backend.connect().getSnapshot('books', 'don-quixote', DAY2, function (error, snapshot) {
+        if (error) return done(error);
+        expect(snapshot).to.eql(v1);
+        done();
+      });
+    });
+
+    it('fetches the version from Day 3', function (done) {
+      backend.connect().getSnapshot('books', 'don-quixote', DAY3, function (error, snapshot) {
+        if (error) return done(error);
+        expect(snapshot).to.eql(v2);
+        done();
+      });
+    });
+
+    it('fetches the latest version if the version is undefined', function (done) {
+      backend.connect().getSnapshot('books', 'don-quixote', undefined, function (error, snapshot) {
+        if (error) return done(error);
+        expect(snapshot).to.eql(v2);
+        done();
+      });
+    });
+
+    it('fetches the latest version when the optional version is not provided', function (done) {
+      backend.connect().getSnapshot('books', 'don-quixote', function (error, snapshot) {
+        if (error) return done(error);
+        expect(snapshot).to.eql(v2);
+        done();
+      });
+    });
+
+    it('can call without a callback', function () {
+      backend.connect().getSnapshot('books', 'don-quixote');
+    });
+
+    it('errors if the version is -1', function (done) {
```

Comment:
> returns an empty snapshot when trying to fetch a snapshot before the document existed

I think this is great. :+1: 

For consistency, shouldn't we also return an empty snapshot, if asked for a version before the document existed (negative version)?